### PR TITLE
Tolerant slicing behavior 

### PR DESCRIPTION
**Pull request by nelson-wu**
Split from https://github.com/fsharp/fslang-design/pull/391

### CODE REVIEW

**Comment by dsyme** [comment]
@nelson-wu Please make the PR for a new RFC document in the RFCs folder with new number

**Comment by nelson-wu** [comment]
> 
> 
> @nelson-wu Please make the PR for a new RFC document in the RFCs folder with new number

Oops, rebased off wrong branch

File: RFCs/FS-1077-tolerant-slicing.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+[drawbacks]: #drawbacks
+
+- We'd change the behavior for cases like `L.[0..99999]` and `L.[-1..0]` where they would become valid. If users are catching that exception and performing logic based on that we'd break their code (they shouldn't be doing so anyways). 
+- If you accidentally access an out-of-bounds slice the compiler's not going to warn you about it.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Keep existing behavior for out-of-bounds slicing
+- Follow the C# behavior and throw errors.
```
Comment (cartermp) [inline RFCs/FS-1077-tolerant-slicing.md]:
```suggestion
- Follow the C# behavior and throw errors, which would be a runtime-breaking change.
```

**Comment by dsyme** [comment]
@charlesroddie All in all I think I need to drink the Pythonesque kool-aid here and accept this behavior.

I know why you've been pushing back on this - this may mask bugs by returning data instead of raising exceptions.  But the image processing examples are compelling enough, as is the argument that Python got some ergonomics right here.

I will approve this, but I don't want to completely close of the discussion - I would appreciate to know the degree of your dislike - where do you sit on a scale of 1 (won't use the language anymore) to 10 (like)

**Comment by dsyme** [comment]
@nelson-wu Really great work on writing these RFCs

**Comment by charlesroddie** [comment]
> I would appreciate to know the degree of your dislike

Thanks. It's small. I estimate 1 bug/coder/year being a little harder to find.

> I don't want to completely close of the discussion

This did bypass a language suggestion effectively so I think it would be good practice to have some community discussion in a public place. I can post a quick summary in a linked thread if needed.

**Comment by dsyme** [comment]
> This did bypass a language suggestion effectively so I think it would be good practice to have some community discussion in a public place. I can post a quick summary in a linked thread if needed.

That's true.  @nelson-wu and @cartermp Can you make a tracking suggestion and discussion thread for this RFC.  I'll mark the suggestion as approved-in-principle as soon as you do

**Comment by nelson-wu** [comment]
> 
> 
> > This did bypass a language suggestion effectively so I think it would be good practice to have some community discussion in a public place. I can post a quick summary in a linked thread if needed.
> 
> That's true. @nelson-wu and @cartermp Can you make a tracking suggestion and discussion thread for this RFC. I'll mark the suggestion as approved-in-principle as soon as you do

Sure! I've created the RFC discussion at https://github.com/fsharp/fslang-design/issues/404 and suggestion at https://github.com/fsharp/fslang-suggestions/issues/795 .

**Pull request merged**